pandas/io/excel/_odfreader.py

```diff
@@ -182,6 +183,20 @@ def _get_column_repeat(self, cell) -> int:
 
         return int(cell.attributes.get((TABLENS, "number-columns-repeated"), 1))
 
+    def _parse_odf_time(self, value: str) -> pd.Timestamp:
+        """
+        Helper function to convert ODF variant of ISO 8601 formatted duration
+        "PnYnMnDTnHnMnS" - see https://www.w3.org/TR/xmlschema-2/#duration
+        """
+        parts = re.match(r"^\s*PT\s*(\d+)\s*H\s*(\d+)\s*M\s*(\d+(\.\d+)?)\s*S$", value)
+        if parts is None:
+            raise ValueError(f"Failed to parse ODF time value: {value}")
+        h, m, s = parts.group(1, 2, 3)
```

Comment:
Could you avoid 1 variable names here?

Reply:
I'm sorry, fixed on 0c70cc2a97a38c52922fd0dc6e7553d2a40d7c04